src/main/kotlin/no/skatteetaten/aurora/gobo/GoboGraphQLContextBuilder.kt

```diff
@@ -0,0 +1,46 @@
+package no.skatteetaten.aurora.gobo
+
+import graphql.servlet.GraphQLContext
+import graphql.servlet.GraphQLContextBuilder
+import no.skatteetaten.aurora.gobo.resolvers.KeysDataLoader
+import no.skatteetaten.aurora.gobo.resolvers.KeysDataLoaderFlux
+import no.skatteetaten.aurora.gobo.resolvers.NoCacheBatchDataLoader
+import no.skatteetaten.aurora.gobo.resolvers.NoCacheBatchDataLoaderFlux
+import org.dataloader.DataLoaderRegistry
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Component
+import javax.servlet.http.HttpServletRequest
+import javax.websocket.server.HandshakeRequest
+
+@Component
+class GoboGraphQLContextBuilder(
+    val loaderList: List<KeysDataLoader<*, *>>,
+    val loaderListFlux: List<KeysDataLoaderFlux<*, *>>
+) : GraphQLContextBuilder {
+
+    private val logger: Logger = LoggerFactory.getLogger(GraphQLContextBuilder::class.java)
+
+    override fun build(httpServletRequest: HttpServletRequest?) = createContext()
+
+    override fun build(handshakeRequest: HandshakeRequest?) = createContext()
+
+    override fun build() = createContext()
+
+    private fun createContext(): GraphQLContext {
+        logger.info("Creating new DataLoader instances")
+
+        val registry = DataLoaderRegistry().apply {
+            loaderList.forEach {
+                register(it::class.simpleName, NoCacheBatchDataLoader(it))
+            }
+            loaderListFlux.forEach {
+                register(it::class.simpleName, NoCacheBatchDataLoaderFlux(it))
```

Comment:
Etter vi finner en løsning på dette så syns jeg vi skal bestemme oss for en måte å skrive dataloaderne på, da kan vi forenkle en del av denne koden.

Reply:
Det er mulig å hente ut klassen denne dataloaderen er dataloader for via reflection her. Da slipper man å måtte forholde seg til en navnekonvensjon andre steder i koden.

`                
val loaderFor = dataLoader::class.supertypes
                    .filter { it.classifier == KeysDataLoader::class }
                    .first()
                    .let { it.arguments[1].type!!.arguments[0].type!!.classifier }!! // DataLoader<String, Try<EtEllerAnnet>>
`

Litt usikker på om kompleksiteten er verdt det dog - vi må også skille på dataloadere som ikke er Try også, kanskje?